CorsixTH/Src/run_length_encoder.cpp

```diff
@@ -65,138 +59,120 @@ bool integer_run_length_encoder::initialise(size_t iRecordSize)
     buffer_size = 0;
     buffer_offset = 0;
     buffer = new (std::nothrow) uint32_t[buffer_capacity];
-    if(!buffer)
-        return false;
+    if (!buffer) return false;
 
     output_capacity = iRecordSize * 32;
     output_size = 0;
     output = new (std::nothrow) uint32_t[output_capacity];
-    if(!output)
-        return false;
+    if (!output) return false;
 
     object_size = 0;
     object_copies = 0;
 
     return true;
 }
 
-void integer_run_length_encoder::write(uint32_t iValue)
-{
+void integer_run_length_encoder::write(uint32_t iValue) {
     buffer[(buffer_offset + buffer_size) % buffer_capacity] = iValue;
-    if(++buffer_size == buffer_capacity)
-        flush(false);
+    if (++buffer_size == buffer_capacity) flush(false);
 }
 
-void integer_run_length_encoder::finish()
-{
-    if(buffer_size != 0)
-        flush(true);
+void integer_run_length_encoder::finish() {
+    if (buffer_size != 0) flush(true);
 }
 
-void integer_run_length_encoder::flush(bool bAll)
-{
-    do
-    {
-        if(object_size == 0)
-        {
+void integer_run_length_encoder::flush(bool bAll) {
+    do {
+        if (object_size == 0) {
             // Decide on the size of the next object
             // Want the object size which gives most object repeats, then for
             // two sizes with the same repeat count, the smaller size.
             size_t iBestRepeats = 0;
             size_t iBestSize = 0;
             size_t iBestOffset = 0;
-            for(size_t iNumRecords = 1; iNumRecords <= 8; ++iNumRecords)
-            {
-                for(size_t iOffset = 0; iOffset < iNumRecords; ++iOffset)
-                {
+            for (size_t iNumRecords = 1; iNumRecords <= 8; ++iNumRecords) {
+                for (size_t iOffset = 0; iOffset < iNumRecords; ++iOffset) {
                     size_t iNumRepeats = 0;
                     size_t iObjSize = iNumRecords * record_size;
-                    while(iObjSize * (iOffset + iNumRepeats + 1) <= buffer_size
-                        && are_ranges_equal(0, iNumRepeats, iOffset, iObjSize))
-                    {
+                    while (iObjSize * (iOffset + iNumRepeats + 1) <=
+                                   buffer_size &&
```

Comment:
Really weird breaking point.

Reply:
Doesn't look weird to me if buffer_size didn't fit on the line, it's the least connected to what's before it, and the expressions left and right of the && are even less connected.

That said, again, a well named function (or functor, or lamba) for the conditional would help readability.

Reply:
Doesn't look weird to me if buffer_size didn't fit on the line, it's the least connected to what's before it, and the expressions left and right of the && are even less connected.

That said, again, a well named function (or functor, or lamba) for the conditional would help readability.